building hip-python from sources fails

Attempted to build hip-python from sources following https://github.com/ROCmSoftwarePlatform/hip-python#build-from-source with

```bash
./build_hip_python_pkgs.sh --pre-clean --no-cuda --no-docs --no-api-docs --no-clean-docs --no-archive
```

This resulted in some errors such as

```
./hip/chip.c:25019:123: error: parameter 2 (‘__pyx_v_attr’) has incomplete type
 static enum hipError_t __pyx_f_3hip_4chip_hipDeviceGetGraphMemAttribute(int __pyx_v_device, enum hipGraphMemAttributeType __pyx_v_attr, void *__pyx_v_value) {
                                                                                                                           ^~~~~~~~~~~~
./hip/chip.c: In function ‘__pyx_f_3hip_4chip_hipDeviceGetGraphMemAttribute’:
./hip/chip.c:25047:156: error: type of formal parameter 2 is incomplete
   __pyx_t_1 = ((enum hipError_t (*)(int, enum hipGraphMemAttributeType, void *))__pyx_v_3hip_4chip__hipDeviceGetGraphMemAttribute__funptr)(__pyx_v_device, __pyx_v_attr, __pyx_v_value); if (unlikely(__Pyx_ErrOccurredWithGIL())) __PYX_ERR(0, 6931, __pyx_L1_error)
```

and

```
./hip/chip.c: At top level:
./hip/chip.c:25208:63: error: unknown type name ‘hipUserObject_t’; did you mean ‘hipTextureObject_t’?
 static enum hipError_t __pyx_f_3hip_4chip_hipUserObjectCreate(hipUserObject_t *__pyx_v_object_out, void *__pyx_v_ptr, hipHostFn_t __pyx_v_destroy, unsigned int __pyx_v_initialRefcount, unsigned int __pyx_v_flags) {
                                                               ^~~~~~~~~~~~~~~
                                                               hipTextureObject_t
```

What could cause this? Maybe the installed rocm is too old? Is there a minimum version of rocm required such that hip-python can be built?

--- a/LICENSE
+++ b/LICENSE
@@ -1,13 +1,21 @@
- * Copyright (c) 2023 Advanced Micro Devices, Inc. All Rights Reserved.
+MIT License
 
-   Licensed under the Apache License, Version 2.0 (the "License");
-   you may not use this file except in compliance with the License.
-   You may obtain a copy of the License at
+Copyright (c) 2023 Advanced Micro Devices, Inc.
 
-       http://www.apache.org/licenses/LICENSE-2.0
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
 
-   Unless required by applicable law or agreed to in writing, software
-   distributed under the License is distributed on an "AS IS" BASIS,
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License.
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.

--- a/README.md
+++ b/README.md
@@ -46,29 +46,35 @@ in the repository's subfolders `hip-python` and `hip-python-as-cuda`.
 
 ## Install Prebuilt Package(s)
 
+<!--
 > **NOTE**: The prebuilt packages might not be available on PyPI directly after a ROCm release as this project is not an official part of the ROCm HIP SDK yet and thus is not fully integrated into the global ROCm HIP SDK build process. Check the `simple` lists to see if your operating system and Python version is supported: [hip-python](https://test.pypi.org/simple/hip-python/), [hip-python-as-cuda](https://test.pypi.org/simple/hip-python-as-cuda/).
+-->
 
 > **NOTE**: Prebuilt packages for some ROCm releases are published to Test PyPI first. Check the `simple` lists to see if your operating system and Python version is supported: [hip-python](https://test.pypi.org/simple/hip-python/), [hip-python-as-cuda](https://test.pypi.org/simple/hip-python-as-cuda/).
 
-### Via PyPI
+> **Warning**: Currently, we have not uploaded any HIP Python packages to PyPI yet. So far we have only uploaded packages to TestPyPI, mainly intended for internal testing purposes. If you find similar named packages on PyPI they may been provided by others, possibly with malicious intent.
+
+### Via TestPyPI
 
 First identify the first three digits of the version number of your ROCm&trade; installation.
 Then install the HIP Python package(s) as follows:
 
 ```shell
-python3 -m pip install hip-python>=$rocm_version
+python3 -m pip install -i https://test.pypi.org/simple hip-python>=$rocm_version
 # if you want to install the CUDA Python interoperability package too, run:
-python3 -m pip install hip-python-as-cuda>=$rocm_version
+python3 -m pip install -i https://test.pypi.org/simple hip-python-as-cuda>=$rocm_version
 ```
 
-#### Via TestPyPI
-
-Packages can be installed via the TestPyPI index by prefixing the
-the PIP install commands as follows:
-
-```shell
-python3 -m pip install -i https://test.pypi.org/simple ...
-```
+<!--
+-- #### Via TestPyPI
+-- 
+-- Packages can be installed via the TestPyPI index by prefixing the
+-- the PIP install commands as follows:
+-- 
+-- ```shell
+-- python3 -m pip install -i https://test.pypi.org/simple ...
+-- ```
+-->
 
 ### Via Wheel in Local Filesystem
 

--- a/README.md
+++ b/README.md
@@ -85,6 +85,8 @@ python3 -m pip install <path/to/hip_python_as_cuda>.whl
 
 ## Build From Source
 
+> **NOTE**: Main is typically based on the latest `release/rocm-rel-X.Y[.Z]` release branch. However, it is recommended to use the release branches directly when building from source.
+
 1. Install ROCM
 1. Install `pip`, virtual environment and development headers for Python 3:
    ```bash

--- a/README.md
+++ b/README.md
@@ -28,14 +28,27 @@ for HIP and an interoperability layer for CUDA&reg; Python programs
 
 ## Requirements
 
-Requires that ROCm&trade; is installed on your system.
+* Currently, only Linux is supported (prebuilt packages and code).
+  * Prebuilt packages distributed via PyPI (or Test PyPI) are only provided for Linux systems that agree with the `manylinux_2_17_x86_64` tag.
+* Requires that a compatible ROCm&trade; HIP SDK is installed on your system.
+  * Source code is provided only for particular ROCm versions.
+    * See the git branches tagged with `release/rocm-rel-X.Y[.Z]`
+  * Prebuilt packages are built only for particular ROCm versions. 
 
-All Python requirements are taking care of by installation scripts. 
+> **NOTE**: You may find that packages for one ROCm&trade; release might be compatible with the ROCm&trade; HIP SDK of another release as the HIP Python functions load HIP C functions in a lazy manner.
+
+### Build requirements
+
+* All Python requirements are taking care of by installation scripts. 
 If you decide not to use these scripts, take a look into the `requirements.txt` file 
 in the top-level folder of the this repository and those 
 in the repository's subfolders `hip-python` and `hip-python-as-cuda`.
 
 ## Install Prebuilt Package(s)
+
+> **NOTE**: The prebuilt packages might not be available on PyPI directly after a ROCm release as this project is not an official part of the ROCm HIP SDK yet and thus is not fully integrated into the global ROCm HIP SDK build process. Check the `simple` lists to see if your operating system and Python version is supported: [hip-python](https://test.pypi.org/simple/hip-python/), [hip-python-as-cuda](https://test.pypi.org/simple/hip-python-as-cuda/).
+
+> **NOTE**: Prebuilt packages for some ROCm releases are published to Test PyPI first. Check the `simple` lists to see if your operating system and Python version is supported: [hip-python](https://test.pypi.org/simple/hip-python/), [hip-python-as-cuda](https://test.pypi.org/simple/hip-python-as-cuda/).
 
 ### Via PyPI
 
@@ -46,6 +59,15 @@ Then install the HIP Python package(s) as follows:
 python3 -m pip install hip-python>=$rocm_version
 # if you want to install the CUDA Python interoperability package too, run:
 python3 -m pip install hip-python-as-cuda>=$rocm_version
+```
+
+#### Via TestPyPI
+
+Packages can be installed via the TestPyPI index by prefixing the
+the PIP install commands as follows:
+
+```shell
+python3 -m pip install -i https://test.pypi.org/simple ...
 ```
 
 ### Via Wheel in Local Filesystem
@@ -69,6 +91,7 @@ python3 -m pip install <path/to/hip_python_as_cuda>.whl
    # Ubuntu:
    sudo apt install python3-pip python3-venv python3-dev
    ```
+1. Check out the feature branch `release/rocm-rel-X.Y[.Z]` for your particular ROCm&trade; installation:
 1. Then run:
    ```bash
    ./build_hip_python_pkgs.sh --post-clean
@@ -134,7 +157,7 @@ For more details on `auditwheel` and how to install it, see: https://pypi.org/pr
 
 ### Auditwheel: Example Output
 
-On Ubuntu 20.04.6 LTS, we obtained for `hip-python`:
+On Ubuntu 20.04.6 LTS, we obtained for `hip-python` using Python 3.8:
 
 ```
 $ python3 -m auditwheel show hip-python/dist/hip_python-5.6.31061.216-cp38-cp38-linux_x86_64.whl
@@ -174,7 +197,7 @@ Therefore, we can use the `manylinux_2_17_x86_64` tag for both packages.
 
 ## Known Compilation Issues
 
-### The `hipsparse` module won't compile with older GCC version
+### The `hipsparse` Module won't Compile with Older GCC Release
 
 With all ROCm&trade; versions before version 5.6.0 (exclusive) and older GCC versions, 
 compiling HIP Python's `hipsparse` module results in a compiler error caused by lines such as:
@@ -183,7 +206,7 @@ compiling HIP Python's `hipsparse` module results in a compiler error caused by 
 HIPSPARSE_ORDER_COLUMN [[deprecated("Please use HIPSPARSE_ORDER_COL instead")]] = 1,
 ```
 
-#### Workaround 1: Disable Build of Hiprand Module
+#### Workaround 1: Disable Build of 'hipsparse' Module
 
 Disabling the build of the `hipsparse` HIP python module can, e.g., 
 be achieved by supplying `--libs "^hipsparse"` to `build_hip_python_pkgs.sh`.

--- a/README.md
+++ b/README.md
@@ -52,7 +52,7 @@ in the repository's subfolders `hip-python` and `hip-python-as-cuda`.
 
 > **NOTE**: Prebuilt packages for some ROCm releases are published to Test PyPI first. Check the `simple` lists to see if your operating system and Python version is supported: [hip-python](https://test.pypi.org/simple/hip-python/), [hip-python-as-cuda](https://test.pypi.org/simple/hip-python-as-cuda/).
 
-> **Warning**: Currently, we have not uploaded any HIP Python packages to PyPI yet. So far we have only uploaded packages to TestPyPI, mainly intended for internal testing purposes. If you find similar named packages on PyPI they may been provided by others, possibly with malicious intent.
+> **WARNING**: Currently, we have not uploaded any HIP Python packages to PyPI yet. So far we have only uploaded packages to TestPyPI, mainly intended for internal testing purposes. If you find similar named packages on PyPI they may been provided by others, possibly with malicious intent.
 
 ### Via TestPyPI
 

--- a/hip-python/docs/user_guide/0_install.md
+++ b/hip-python/docs/user_guide/0_install.md
@@ -71,6 +71,7 @@ Both assumptions often apply.
 
 ### Installation Commands
 
+
 :::{important}
 
 Especially on older operating systems, ensure that your `pip` is upgraded to
@@ -84,7 +85,7 @@ python3 -m pip install --upgrade pip
 After having identified the correct package for your ROCm&trade; installation, type:
 
 ```shell
-python3 -m pip install hip-python-<hip_version>.<hip_python_version>
+python3 -m pip install -i https://test.pypi.org/simple hip-python>=<rocm_version>.<hip_python_version>
 ```
 
 or if you have a HIP Python wheel somewhere in your filesystem:
@@ -92,6 +93,14 @@ or if you have a HIP Python wheel somewhere in your filesystem:
 ```shell
 python3 -m pip install <path/to/hip_python>.whl
 ```
+:::{warning}
+
+Currently, we have not uploaded any HIP Python packages to PyPI yet.
+So far we have only uploaded packages to TestPyPI, mainly intended for internal
+testing purposes. If you find similar named packages
+on PyPI they may been provided by others, possibly with
+malicious intent.
+:::
 
 :::{note}
 

